server/src/weaverbird/pipeline/pipeline.py

```diff
@@ -256,6 +256,63 @@ def remove_void_conditions_from_filter_steps(
     return final_steps
 
 
+def _is_empty(data: dict[str, Any] | list[Any]) -> bool:
+    if isinstance(data, dict):
+        return len(data) == 0
+    elif isinstance(data, list):
+        return len(data) == 0
+    return False
+
+
+def _remove_empty_elements_that_has_empty_value(
```

Comment:
I don't understand the function name... It's supposed to remove emtpy elements right ?

```suggestion
def _remove_empty_elements(
```